### PR TITLE
Contracts: Improve Wallet's AccessControl setup

### PR DESCRIPTION
**Pull request by facuspagnuolo**
### MERGE AFTER #406

### CODE REVIEW

**Review by hazim-j — COMMENTED** [review]
Looks good but left a comment regarding upgradability.

File: apps/contracts/contracts/ERC4337/Wallet.sol [hunk 1]
```diff
@@ -31,18 +31,26 @@ contract Wallet is
...
-  // 3. Deprecated variables cannot be deleted and should be marked instead.
+  // solhint-disable var-name-mixedcase
+  bytes32 public constant OWNER_ROLE = keccak256("OWNER_ROLE");
+  bytes32 public constant GUARDIAN_ROLE = keccak256("GUARDIAN_ROLE");
+  // solhint-enable var-name-mixedcase
+
   uint256 public nonce;
   address public entryPoint;
-  bytes32 public OWNER_ROLE; // solhint-disable-line var-name-mixedcase
-  bytes32 public GUARDIAN_ROLE; // solhint-disable-line var-name-mixedcase
```
Comment (hazim-j) [inline apps/contracts/contracts/ERC4337/Wallet.sol:42]: _See above comment_

File: apps/contracts/contracts/ERC4337/Wallet.sol [hunk 2]
```diff
@@ -31,18 +31,26 @@ contract Wallet is
...
-  // This contract is an implementation for WalletProxy.sol.
-  // The following must be followed when updating contract variables:
-  // 1. Order and type of variables cannot change.
-  // 2. New variables must be added last.
-  // 3. Deprecated variables cannot be deleted and should be marked instead.
+  // solhint-disable var-name-mixedcase
+  bytes32 public constant OWNER_ROLE = keccak256("OWNER_ROLE");
+  bytes32 public constant GUARDIAN_ROLE = keccak256("GUARDIAN_ROLE");
+  // solhint-enable var-name-mixedcase
+
```
[the remark this answers is not in the document]
Comment (facuspagnuolo) [reply apps/contracts/contracts/ERC4337/Wallet.sol:38]: No, these are constant variables, not storage variables. Upgradeability can be affected if you restructure your storage layout, constant variables are translated to pure functions by the compiler

[the remark this answers is not in the document]
Comment (hazim-j) [reply apps/contracts/contracts/ERC4337/Wallet.sol:38]: Interesting. Didn’t know that. What would be an example of a storage variable that can affect upgradability? I was under the impression that the nonce, entryPoint, OWNER_ROLE, and GUARDIAN_ROLE variables below took up storage slots? Or do they also get converted to pure functions?

[the remark this answers is not in the document]
Comment (facuspagnuolo) [reply apps/contracts/contracts/ERC4337/Wallet.sol:38]: `nonce` and `entryPoint` are examples of storage variables, `OWNER_ROLE` and `GUARDIAN_ROLE` are constants. 

Constants are not stored in the contract, are simply pure functions, you don't need to perform a SLOAD to fetch their values. On the other hand, `nonce` and `entryPoint` you do need to perform an SLOAD, you're accessing the storage mapping of the contract in that case.

**Pull request merged**